src/legacy/core_plugins/vis_default_editor/public/components/agg.tsx

```diff
@@ -114,6 +120,13 @@ function DefaultEditorAgg({
     }
   }
 
+  // This useEffect is required to initiate rerender to keep labels up to date (Issue #57822).
+  useEffect(() => {
+    if (aggName === 'date_histogram') {
+      setLocalTimeRange(timeRange);
+    }
+  }, [aggName, timeRange]);
```

Comment:
Those "never consumed states" that are just meant to trigger a rerender of the component, are very prone to be refactored away accidentally one day. I think we could improve this, by simply storing the `aggDescription` in a state variable and update that from the hook instead. Untested example code:

```ts
const buildAggDescription = () => {
  let aggDescription = '';
  if (agg.type && agg.type.makeLabel) {
    try {
      aggDescription = agg.type.makeLabel(agg);
    } catch () {
      // we actually don't need to reset to empty string here, it's anyway an empty string in case we ever come here.
    }
  }
  return aggDescription;
};
const [aggDescription, setAggDescription] = useState(buildAggDescription());

useEffect(() => {
  if (agg.type.name === 'date_histogram') {
    agg.aggTypes.setTimeRange(timeRange);
    setAggDescription(buildAggDescription());
  }
}, [agg, timeRange]);
```

That way we could avoid some empty state variable we never actually consume (and also won't end up with more code).